spec/unit/storage_spec.rb

```diff
@@ -307,4 +307,25 @@ def check_file(file)
       storage.upload_file('assets/some_longer_path/local_image2.jpg')
     end
   end
+
+  describe '#delete_extra_remote_files' do
+    it 'should delete the files in bulk' do
+      remote_files = ['public/image.png']
+
+      storage = AssetSync::Storage.new(@config)
+
+      [:local_files, :ignored_files, :always_upload_files].each do |method|
+         expect(storage).to receive(method).and_return([])
+      end
+
+      expect(storage).to receive(:get_remote_files).and_return(remote_files)
+      expect(storage).to receive(:connection).and_return(connection = double)
```

Comment:
Maybe can assign these variable above?
Inline assignments are easier to be overlook